service/src/main/java/tds/session/repositories/ExternRepository.java

```diff
@@ -0,0 +1,14 @@
+package tds.session.repositories;
+
+import java.util.Optional;
+
+import tds.session.Extern;
+
+public interface ExternRepository {
+    /**
+     * Fetches the extern by given client name
+     * @param clientName client name for extern
+     * @return optional containing {@link tds.session.Extern Extern} otherwise empty Optional
+     */
+    Optional<Extern> getExternByClientName(String clientName);
```

Comment:
We can discuss @millerts example today.

The comment on what @jeffjohnson9046 said, I think that is a good point.  Maybe it does make sense to keep the pattern for consistency but also because the caller might be able to come up with an intelligent default configuration to use.